src/codec/crypto/authentication_tag.rs

```diff
@@ -47,6 +47,11 @@ impl Deref for AuthenticationTag {
     }
 }
 
+impl Default for AuthenticationTag {
+    fn default() -> Self {
+        Self([0u8; TAG_LENGTH])
+    }
+}
```

Comment:
I'm not sure it makes sense to create Default / all zeros for these data structures. They really only have meaning when they are over some specific data. I haven't gotten far enough into the PR but I'm guessing these are for tests? I'd rather see a helper function in the test that generates these all zero tags if they're needed there rather than exposing them through the public API.